### PR TITLE
Distinguish between upcoming and past events.

### PR DESCRIPTION
**Pull request by sarahboyce**
https://staging-djangonaut-space.azurewebsites.net/events/

image

### CODE REVIEW

File: home/templates/home/prerelease/event_list.html [hunk 1]
```diff
@@ -27,10 +27,29 @@ <h3>Events</h3>
             {% endif %}
             </div>
         </div>
+        <h2>Upcoming Events</h2>
         <div class="g-2 g-lg-3 row row-cols-4 justify-content-evenly pt-4">
```
Comment (sabderemane) [inline home/templates/home/prerelease/event_list.html:31]:
```suggestion
        <div class="g-2 g-lg-3 row justify-content-evenly pt-4">
```
This look very small with this property in mobile. If there is already the class `col` in each element in `event_card.html`, it's should be fine normally :)

File: home/templates/home/prerelease/event_list.html [hunk 2]
```diff
@@ -27,10 +27,29 @@ <h3>Events</h3>
...
 
-            {% for event in events %}
+            {% for event in upcoming_events %}
             {% include 'home/includes/event_card.html' %}
+            {% empty %}
+            <div class="text-center">
+                <img alt="No data image" class="no-data-img" src="{% static 'img/no_data.svg' %}" />
+                <p class="pt-4">No upcoming events.</p>
```
Comment (sabderemane) [inline home/templates/home/prerelease/event_list.html:38]:
```suggestion
                <p class="pt-4 pb-4">No upcoming events.</p>
```
On mobile, it's too close to the next title without padding

File: home/templates/home/prerelease/event_list.html [hunk 3]
```diff
@@ -27,10 +27,29 @@ <h3>Events</h3>
...
+        <h2>Past Events</h2>
+        <div class="g-2 g-lg-3 row row-cols-4 justify-content-evenly pt-4">
+
+            {% for event in past_events %}
+            {% include 'home/includes/event_card.html' %}
+            {% empty %}
+            <div class="text-center">
+                <img alt="No data image" class="no-data-img" src="{% static 'img/no_data.svg' %}" />
+                <p class="pt-4">No past events.</p>
+            </div>
```
Comment (sabderemane) [inline home/templates/home/prerelease/event_list.html:52]: ♻️  Same than the previous comment

**Review by sabderemane — COMMENTED** [review]
LGTM 🚀  I added few comments to improve the responsive :)

File: home/views.py [hunk 4]
```diff
@@ -56,13 +56,16 @@ class EventListView(ListView):
...
 
         tag = self.request.GET.get("tag")
         if tag:
-            events = events.filter(tags__name=tag)
+            upcoming_events = upcoming_events.filter(tags__name=tag)
+            past_events = past_events.filter(tags__name=tag)
 
-        context["events"] = events
+        context["upcoming_events"] = upcoming_events
+        context["past_events"] = past_events
```
Comment (tim-schilling) [inline home/views.py:68]: We could use the base `events` queryset, then apply the datetime filtering just before it's needed:

```suggestion
        context["upcoming_events"] = events.upcoming()
        context["past_events"] = events.past()
```

Comment (sarahboyce) [reply home/views.py]: Thank you ⭐

**Comment by sarahboyce** [comment]
Thank you @sabderemane - looks much better now (I will redeploy shortly)
image

**Pull request merged**